Tests/iaas/entropy/entropy-check.py

```diff
@@ -402,38 +402,60 @@ def handle(self, record):
         self.bylevel[record.levelno] += 1
 
 
-def _deduce_version(name, ubuntu_ver=re.compile(r"\d\d\.\d\d\Z"), debian_ver=re.compile(r"\d+\Z")):
-    """helper for `select_deb_image` to deduce a version even if its only given via codename"""
-    canonicalized = [part.strip() for part in name.lower().split()]
-    if "debian" in canonicalized:
-        # don't even consider "stretch" (9) here
-        codenames = ("buster", "bullseye", "bookworm")
-        for idx, name in enumerate(codenames):
-            if name in canonicalized:
-                return idx + 10
-        for part in canonicalized:
-            if debian_ver.match(part):
-                return int(part)
-    elif "ubuntu" in canonicalized:
-        for part in canonicalized:
-            if ubuntu_ver.match(part):
-                return int(part[:2] + part[3:])
-    return -1
+# only list stable versions here
+DEBIAN_CODENAMES = {
+    "buster": 10,
+    "bullseye": 11,
+    "bookworm": 12,
+}
+
+
+def _deduce_deb_version(name, debian_ver=re.compile(r"\d+\Z")):
+    name = name.lower().strip()
+    if debian_ver.match(name):
+        return int(name)
+    return DEBIAN_CODENAMES.get(name, 0)
+
+
+def _deduce_ubu_version(name, ubuntu_ver=re.compile(r"\d\d\.\d\d\Z")):
+    name = name.lower().strip()
+    if ubuntu_ver.match(name):
+        return int(name[:2] + name[3:])
```

Comment:
What about just removing the dot from the string instead of cutting and rejoining it? I makes it pretty hard to read

Reply:
f6ebaebcad4a0ee88636c79c09ed8ce8fb85c731